specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2021-01-28-preview/HybridCompute.json

```diff
@@ -832,6 +832,17 @@
         "privateLinkScopeResourceId": {
           "type": "string",
           "description": "The resource id of the private link scope this machine is assigned to, if any."
+        },
+        "parentClusterResourceId": {
+          "type": "string",
+          "description": "The resource id of the parent cluster (Azure HCI) this machine is assigned to, if any."
+        },
+        "detectedProperties": {
+          "type": "object",
```

Comment:
Would be good to strongly type 'detectedProperties' and include as a field in the 'definitions' section of the swagger

Reply:
can we strongly type dictionary<string, string> in the swagger?

Reply:
Hi @dorothyhu - you can create a new type. Look at Extensions as an example: $ref can point to a type

  {
            "name": "extensionParameters",
            "in": "body",
            "required": true,
            "schema": {
              "$ref": "#/definitions/MachineExtensionUpdate"
            },
            "description": "Parameters supplied to the Create Machine Extension operation."
          },